From: "Mike Rapoport (Microsoft)" <rppt@kernel.org>
Subject: sh: remove setup_bootmem_node() and plat_mem_setup()
Date: Sun, 10 May 2026 16:55:40 +0300

setup_bootmem_node() and plat_mem_setup() were needed to setup NUMA
"nodes" for URAM memory ranges.  Since there is no NUMA support anymore,
these functions are not needed.

Remove them.

Link: https://lore.kernel.org/20260510135546.13554-5-rppt@kernel.org
Signed-off-by: Mike Rapoport (Microsoft) <rppt@kernel.org>
Reviewed-by: Arnd Bergmann <arnd@arndb.de>
Cc: John Paul Adrian Glaubitz <glaubitz@physik.fu-berlin.de>
Cc: Rich Felker <dalias@libc.org>
Cc: Yoshinori Sato <ysato@users.sourceforge.jp>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 arch/sh/kernel/cpu/sh4a/setup-sh7366.c |    5 -----
 arch/sh/kernel/cpu/sh4a/setup-sh7722.c |    6 ------
 arch/sh/kernel/cpu/sh4a/setup-sh7757.c |    4 ----
 arch/sh/kernel/cpu/sh4a/setup-sh7785.c |    6 ------
 arch/sh/kernel/cpu/sh4a/setup-sh7786.c |    4 ----
 arch/sh/kernel/cpu/sh4a/setup-shx3.c   |   17 -----------------
 arch/sh/mm/init.c                      |    7 -------
 7 files changed, 49 deletions(-)

--- a/arch/sh/kernel/cpu/sh4a/setup-sh7366.c~sh-remove-setup_bootmem_node-and-plat_mem_setup
+++ a/arch/sh/kernel/cpu/sh4a/setup-sh7366.c
@@ -381,8 +381,3 @@ void __init plat_irq_setup(void)
 {
 	register_intc_controller(&intc_desc);
 }
-
-void __init plat_mem_setup(void)
-{
-	/* TODO: Register Node 1 */
-}
--- a/arch/sh/kernel/cpu/sh4a/setup-sh7722.c~sh-remove-setup_bootmem_node-and-plat_mem_setup
+++ a/arch/sh/kernel/cpu/sh4a/setup-sh7722.c
@@ -658,9 +658,3 @@ void __init plat_irq_setup(void)
 {
 	register_intc_controller(&intc_desc);
 }
-
-void __init plat_mem_setup(void)
-{
-	/* Register the URAM space as Node 1 */
-	setup_bootmem_node(1, 0x055f0000, 0x05610000);
-}
--- a/arch/sh/kernel/cpu/sh4a/setup-sh7757.c~sh-remove-setup_bootmem_node-and-plat_mem_setup
+++ a/arch/sh/kernel/cpu/sh4a/setup-sh7757.c
@@ -1239,7 +1239,3 @@ void __init plat_irq_setup_pins(int mode
 		BUG();
 	}
 }
-
-void __init plat_mem_setup(void)
-{
-}
--- a/arch/sh/kernel/cpu/sh4a/setup-sh7785.c~sh-remove-setup_bootmem_node-and-plat_mem_setup
+++ a/arch/sh/kernel/cpu/sh4a/setup-sh7785.c
@@ -600,9 +600,3 @@ void __init plat_irq_setup_pins(int mode
 		BUG();
 	}
 }
-
-void __init plat_mem_setup(void)
-{
-	/* Register the URAM space as Node 1 */
-	setup_bootmem_node(1, 0xe55f0000, 0xe5610000);
-}
--- a/arch/sh/kernel/cpu/sh4a/setup-sh7786.c~sh-remove-setup_bootmem_node-and-plat_mem_setup
+++ a/arch/sh/kernel/cpu/sh4a/setup-sh7786.c
@@ -783,10 +783,6 @@ void __init plat_irq_setup_pins(int mode
 	}
 }
 
-void __init plat_mem_setup(void)
-{
-}
-
 static int __init sh7786_devices_setup(void)
 {
 	int ret, irq;
--- a/arch/sh/kernel/cpu/sh4a/setup-shx3.c~sh-remove-setup_bootmem_node-and-plat_mem_setup
+++ a/arch/sh/kernel/cpu/sh4a/setup-shx3.c
@@ -377,20 +377,3 @@ void __init plat_irq_setup(void)
 {
 	register_intc_controller(&intc_desc);
 }
-
-void __init plat_mem_setup(void)
-{
-	unsigned int nid = 1;
-
-	/* Register CPU#0 URAM space as Node 1 */
-	setup_bootmem_node(nid++, 0x145f0000, 0x14610000);	/* CPU0 */
-
-#if 0
-	/* XXX: Not yet.. */
-	setup_bootmem_node(nid++, 0x14df0000, 0x14e10000);	/* CPU1 */
-	setup_bootmem_node(nid++, 0x155f0000, 0x15610000);	/* CPU2 */
-	setup_bootmem_node(nid++, 0x15df0000, 0x15e10000);	/* CPU3 */
-#endif
-
-	setup_bootmem_node(nid++, 0x16000000, 0x16020000);	/* CSM */
-}
--- a/arch/sh/mm/init.c~sh-remove-setup_bootmem_node-and-plat_mem_setup
+++ a/arch/sh/mm/init.c
@@ -38,11 +38,6 @@ void __init generic_mem_init(void)
 	memblock_add(__MEMORY_START, __MEMORY_SIZE);
 }
 
-void __init __weak plat_mem_setup(void)
-{
-	/* Nothing to see here, move along. */
-}
-
 #ifdef CONFIG_MMU
 static pte_t *__get_pte_phys(unsigned long addr)
 {
@@ -209,8 +204,6 @@ static void __init do_init_bootmem(void)
 		__add_active_range(0, start_pfn, end_pfn);
 
 	node_set_online(0);
-
-	plat_mem_setup();
 }
 
 static void __init early_reserve_mem(void)
_
